### PR TITLE
test: Re-enable tests on windows.

### PR DESCRIPTION
**Pull request by iphydf**
Some of these aren't working still.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/qTox/154)

### CODE REVIEW

**Comment by nurupo** [comment]
What do you mean "can't find Qt plugin DLLs"? Just put the needed plugin dlls in appropriate directories alongside the `.exe` file. Plugin dlls typically have their own directories, e.g. platforms, imageformats, iconengines, etc. - you can see a qtox windows release zip having those.

**Comment by iphydf** [comment]
> What do you mean "can't find Qt plugin DLLs"? Just put the needed plugin dlls in appropriate directories alongside the `.exe` file. Plugin dlls typically have their own directories, e.g. platforms, imageformats, iconengines, etc. - you can see a qtox windows release zip having those.
> 
> Also, you might need to enable building of the specific plugin in the Windows Qt cross-compilation config if the Qt plugin dll is not being built in the first place.

What I mean is that the tests running on CI don't find the plugins because they don't know that we magically put them into `/export`. Our setup and zip do know that because we coded that into the setup/zip script.

File: windows/cross-compile/build.sh [hunk 1]
```diff
@@ -106,11 +106,12 @@ cp -r /export/* "$QTOX_PREFIX_DIR"
...
 if [[ $RUN_TESTS -ne 0 ]]; then
-  export WINEPATH='/export;/windows/bin'
+  export WINEQT_QPA_PLATFORM='offscreen'
+  export WINEQT_PLUGIN_PATH='z:\export'
+  export WINEPATH='z:\export;z:\windows\bin'
   export CTEST_OUTPUT_ON_FAILURE=1
   export PATH="$PATH:/opt/wine-stable/bin"
-  # TODO(iphydf): Fix tests on windows.
-  # ctest -j$(nproc)
+  ctest -j"$(nproc)"
```
Comment (Green-Sky) [inline windows/cross-compile/build.sh:114]: --output-on-failure

Comment (iphydf) [reply windows/cross-compile/build.sh]: Shouldn't make a difference vs. the envvar but sure. I agree it's nicer this way.

**Comment by nurupo** [comment]
My understanding is that Qt looks for plugins in hardcoded directories alongside the .exe file. So just put the needed plugin dlls in appropriate directories alongside the tests .exe files and the problem is solved, the exe files would be able to find the plugins.

Apparently setting `QT_PLUGIN_PATH` should work too, didn't know of that env variable, that looks like a better way of doing that too rather than copying/symlinking the plugin dirs.

File: cmake/Testing.cmake [hunk 2]
```diff
@@ -34,7 +34,10 @@ auto_test(core toxid "" "")
 auto_test(core toxstring "" "")
 auto_test(core fileprogress "" "")
 auto_test(chatlog textformatter "" "")
-auto_test(net bsu "${${PROJECT_NAME}_RESOURCES}" "") # needs nodes list
+# TODO(iphydf): Fix on windows.
```
Comment (Green-Sky) [inline cmake/Testing.cmake:37]: I thought you fixed this?

Comment (iphydf) [reply cmake/Testing.cmake]: Fixed.

**Review by Green-Sky — APPROVED** [review]

**Pull request merged**